cirq-core/cirq/qis/clifford_tableau.py

```diff
@@ -176,6 +178,43 @@ def _str_full_(self) -> str:
 
         return string
 
+    def merged_with(self, second: 'CliffordTableau') -> 'CliffordTableau':
```

Comment:
How about making this `__mul__` (or `__rmul__` if you want to reverse the order)? It might be nice to be able to use `*` to multiply two tableaus

Reply:
I have another PR that merges (and unmerges) states. We should be consistent with names.

Reply:
In that case, are you doing the tensor product? If yes, we shouldn't use "merge" for either!

Reply:
Let me give more background information.  The reason I choose `merge_with` is simple -- because it is used in `SingleCliffordGate`, see [here](https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/ops/clifford_gate.py#L285). The meaning of this method is simply a conjugate mapping of two functions `U_2U_1 g U_1'U_2'`. (Or equivalently, just composed function if viewing in state vector;) I think it is different from the meaning of merge_state as @daxfohl mentioned.

As for @smitsanghavi  mentioned `__mul__` or other magic functions, the benefits of doing this are less in here. Because this is the helper function, I plan to use that in the CliffordGate. So personally, I prefer the `compose_with` so far. Anyway, we can discuss that in cirq-cync meeting?

Reply:
for the record: we discussed in cirq-cync; had many votes; and settled on `then` as the method name and supporting `@` for matrix-multiply-like